frontend/src/pages/GraphicalSelector/GraphicalSelector.jsx

```diff
@@ -1,72 +1,132 @@
 import React, { useEffect, useRef, useState } from "react";
-import { findDOMNode } from "react-dom";
 import { useSelector } from "react-redux";
 import G6 from "@antv/g6";
 import axios from "axios";
 import PageTemplate from "components/PageTemplate";
+import S from "./styles";
 
 const GraphicalSelector = () => {
   const { programCode, specs } = useSelector((state) => state.degree);
 
-  const [courses, setCourses] = useState([]);
-  const [courseEdges, setCourseEdges] = useState([]);
+  const [graph, setGraph] = useState(null);
+  const [loading, setLoading] = useState(true);
```

Comment:
I feel like react had a better way to do this now, but I forget what it was

Reply:
Was it this https://reactjs.org/docs/react-api.html#reactsuspense? Doesn't seem like they support fetching yet :(